File: Desktop/View/Table/Header/SortableButton.cs
using System.Windows.Controls;
using Desktop.Include;

namespace Desktop.View.Table.Header
{
    internal static class SortableButtonExtensions
    {
        public static string ToString(this SortableButton.ButtonState state, object _)
            => state == SortableButton.ButtonState.None ? "" : state.ToString();
    }
    public class SortableButton : Button
    {
        public enum ButtonState : short
        {
            None = 0, Asc = 1, Desc = 2
        }
        
        private ButtonState _state = ButtonState.None;
        public string ColumnTitle = "UUID";

        public ButtonState State
        {
            get => _state;
            set
            {
                _state = value;
                string field;
                if ((field = State.ToString(true)) != "")
                    Content = ColumnTitle + " " + field;
                else
                    Content = ColumnTitle;
                Content = (Content as string).AddSpacesBeforeCapitalized();
            }
        }

        public SortableButton(SortableHeader header)
        {
            header.Add(this);
            Click += (sender, args) => State = (ButtonState)((1 + (short)State) % 3);
            Click += (sender, args) => header.Update(this);
        }
    }
}

File: Mobile/app/src/main/java/com/mauzerov/mobile/views/SortableButton.kt
package com.mauzerov.mobile.views

import android.annotation.SuppressLint
import android.content.Context
import android.util.AttributeSet
import android.util.Log
import android.view.MotionEvent
import android.view.View
import android.widget.Button
import androidx.appcompat.widget.AppCompatButton
import com.mauzerov.mobile.views.SortableButton.ButtonState



class SortableButton(context: Context, attrs: AttributeSet?, defStyleAttr: Int, header : SortableHeader) :
    androidx.appcompat.widget.AppCompatButton(context, attrs) {

    constructor(context: Context, attrs: AttributeSet?, header : SortableHeader) : this(context, attrs, 0, header)
    constructor(context: Context, header : SortableHeader) : this(context, null, 0, header)

    var _header : SortableHeader;

    enum class ButtonState {
        None, Asc, Desc {
            override fun next() : ButtonState { return None };
        };
        open fun next() : ButtonState{
            return values()[ordinal + 1]
        }
    }

    var state = ButtonState.None
        @SuppressLint("SetTextI18n")
        set(value) {
            field = value;
            if (value == ButtonState.None)
                this@SortableButton.text = columnTitle;
            else
                this@SortableButton.text = "$columnTitle $state"
        }
    var columnTitle = "UUID"

    init {
        header += this;
        _header = header;
    }
}